### PR TITLE
convert cli/model to pytest

### PR DESCRIPTION
**Pull request by peterdragun**
Test results:
```
pytest -v tests/foreman/cli/test_model.py
================================== test session starts ===================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 --
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo
plugins: forked-1.0.2, xdist-1.29.0, mock-1.10.4, cov-2.7.1, services-1.3.1
collecting ... 2020-08-20 11:21:56 - conftest - DEBUG - Collected 32 test cases
collected 32 items                                                                       

tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[0] PASSED      [  3%]
tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[1] PASSED      [  6%]
tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[2] PASSED      [  9%]
tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[3] PASSED      [ 12%]
tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[4] PASSED      [ 15%]
tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[5] PASSED      [ 18%]
tests/foreman/cli/test_model.py::TestModel::test_positive_crud_with_name[6] PASSED      [ 21%]
tests/foreman/cli/test_model.py::TestModel::test_positive_create_with_vendor_class PASSED [ 25%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[0] PASSED    [ 28%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[1] PASSED    [ 31%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[2] PASSED    [ 34%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[3] PASSED    [ 37%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[4] PASSED    [ 40%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[5] PASSED    [ 43%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[6] PASSED    [ 46%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[7] PASSED    [ 50%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[8] PASSED    [ 53%]
tests/foreman/cli/test_model.py::TestModel::test_negative_create_with_name[9] PASSED    [ 56%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[0] PASSED         [ 59%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[1] PASSED         [ 62%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[2] PASSED         [ 65%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[3] PASSED         [ 68%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[4] PASSED         [ 71%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[5] PASSED         [ 75%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[6] PASSED         [ 78%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[7] PASSED         [ 81%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[8] PASSED         [ 84%]
tests/foreman/cli/test_model.py::TestModel::test_negative_update_name[9] PASSED         [ 87%]
tests/foreman/cli/test_model.py::TestModel::test_negative_delete_by_id[0] PASSED        [ 90%]
tests/foreman/cli/test_model.py::TestModel::test_negative_delete_by_id[1] PASSED        [ 93%]
tests/foreman/cli/test_model.py::TestModel::test_negative_delete_by_id[2] PASSED        [ 96%]
tests/foreman/cli/test_model.py::TestModel::test_negative_delete_by_id[3] PASSED        [100%]
=========================== 32 passed, 3 warnings in 484.54 seconds ===========================
```

### CODE REVIEW

**Review by tstrych — APPROVED** [review]
ACK, rewritten into pytest :) and create,update,delete was squashed into one test :) 

I have a question for somebody ? Why there is % decrease in `robottelo/decorators/__init__.py` in code coverage. I would be happy if there is someone who can explain to me, how that works :) thanks

**Review by omkarkhatavkar — APPROVED** [review]
ACK, if you think you can merge some of the tests those are having a similar intention. (this is nonblocking suggestion)

**Comment by peterdragun** [comment]
PR was updated to use `:parametrized: yes` in docstring for correct uploading of all parametrized tests to Polarion. 

@omkarkhatavkar Not sure which tests you mean. I already merged almost all positive tests into one crud test. I don't see any benefits in merging negative tests. Am I missing something?

File: tests/foreman/cli/test_model.py [hunk 1]
```diff
@@ -58,88 +75,54 @@ def test_positive_create_with_vendor_class(self):
...
+        with pytest.raises(CLIReturnCodeError):
+            Model.update({'id': class_model['id'], 'new-name': new_name})
+        result = Model.info({'id': class_model['id']})
+        assert class_model['name'] == result['name']
 
     @tier1
-    def test_negative_delete_by_id(self):
+    @pytest.mark.parametrize('entity_id', **parametrized(invalid_id_list()))
+    def test_negative_delete_by_id(self, entity_id):
         """Create Model then delete it by wrong ID
```
Comment (dvagala) [inline tests/foreman/cli/test_model.py:117]: Just pointing out that the comment should probably be updated, since the test doesn't create the Model.

**Review by mirekdlugosz — APPROVED** [review]
LGTM

**Pull request merged**